Script/AtomicWebViewEditor/clientExtensions/languageExtensions/typescript/TypescriptLanguageExtension.ts

```diff
@@ -291,7 +332,41 @@ export default class TypescriptLanguageExtension implements Editor.ClientExtensi
      * @return {[type]}
      */
     preferencesChanged() {
-        // Stub function for now
-        this.serviceLocator.clientServices.getUserPreference("TypescriptLanguageExtension", "CompileOnSave", true);
+        if (this.active) {
+            let compileOnSave = this.serviceLocator.clientServices.getUserPreference("HostTypeScriptLanguageExtension", "CompileOnSave", true);
+            const message: WorkerProcessTypes.SetPreferencesMessageData = {
+                command: WorkerProcessTypes.SetPreferences,
+                preferences: {
+                    compileOnSave: compileOnSave
+                }
+            };
+
+            this.worker.port.postMessage(message);
+        }
+    }
+
+    /**
+     * Tell the language service to perform a full compile
+     */
+    doFullCompile() {
+        if (this.active) {
+            const message: WorkerProcessTypes.FullCompileMessageData = {
+                command: WorkerProcessTypes.DoFullCompile,
+                tsConfig: this.getTsConfig()
+            };
+            this.worker.port.postMessage(message);
+        }
+    }
+
+
+    /**
+     * Displays the results from a full compile
+     * @param  {WorkerProcessTypes.FullCompileResultsMessageData} results
+     */
+    displayFullCompileResults(results: WorkerProcessTypes.FullCompileResultsMessageData) {
+        let messageArray = results.annotations.map((result) => {
+            return `${result.text} at line ${result.row} col ${result.column} in ${result.file}`;
+        });
+        window.atomicQueryPromise("TypeScript.DisplayCompileResults", results);
     }
 }
```

Comment:
Ok, so now we're in the WebView, the consistency between the Atomic Editor scripts and this is pretty darn cool.  We can also use Atomic from the webview with the emscripten stuff, which may open up some opportunities.  Really impressed here, will need to get the Debugger/Inspector back up with new Chromium